2009-05-06  Michael Tautschnig  <mt@debian.org>

	* setup-storage/Parser.pm: Set the boot flag on partition mounted at /, if
		bootable is not explicitly set.
Index: trunk/lib/setup-storage/Parser.pm
===================================================================
--- trunk.orig/lib/setup-storage/Parser.pm
+++ trunk/lib/setup-storage/Parser.pm	
@@ -522,6 +522,12 @@
           } else {
             $FAI::partition_pointer->{encrypt} = 0;
           }
+          if ($FAI::device =~ /^PHY_(.+)$/ &&
+            -1 == $FAI::configs{$FAI::device}{bootable} &&
+            $FAI::partition_pointer->{mountpoint} eq "/") {
+            $FAI::configs{$FAI::device}{bootable} =
+              $FAI::partition_pointer->{number};
+          }
         }
 
     name: m{^([^/,\s\-]+)}
